docs/two-phase-commit-transactions.md

```diff
@@ -193,178 +215,88 @@ tx.commit();
-          // Rolling back the transaction failed. You can log it here
-        }
-      } catch (UnsatisfiedConditionException e) {
-        // You need to handle UnsatisfiedConditionException only if a mutation operation specifies a condition.
-        // This exception indicates the condition for the mutation operation is not met, so you can
-        // retry the transaction once the exception cause is fixed
-        try {
-          tx.rollback();
-        } catch (RollbackException ex) {
-          // Rolling back the transaction failed. You can log it here
-        }
-      } catch (CrudException | PreparationException | ValidationException | CommitException e) {
-        // If you catch CrudException or PreparationException or ValidationException or
-        // CommitException, it indicates some failure happens, so you should cancel the transaction
-        // or retry the transaction after the failure or error is fixed
-        try {
-          tx.rollback();
-        } catch (RollbackException ex) {
-          // Rolling back the transaction failed. You can log it here
-        }
-        return;
-      } catch (UnknownTransactionStatusException e) {
-        // If you catch `UnknownTransactionStatusException` when committing the transaction, you are
-        // not sure if the transaction succeeds or not. In such a case, you need to check if the
-        // transaction is committed successfully or not and retry it if it failed. How to identify a
-        // transaction status is delegated to users
-        return;
-      }
-    }
+TransactionFactory factory1 =
+    TransactionFactory.create("<configuration file path for the transaction manager1>");
+TwoPhaseCommitTransactionManager transactionManager1 =
+    factory1.getTwoPhaseCommitTransactionManager();
+
+TransactionFactory factory2 =
+    TransactionFactory.create("<configuration file path for the transaction manager2>");
+TwoPhaseCommitTransactionManager transactionManager2 =
+    factory2.getTwoPhaseCommitTransactionManager();
+
+TwoPhaseCommitTransaction transaction1 = null;
+TwoPhaseCommitTransaction transaction2 = null;
+try {
+  // Begin a transaction
+  transaction1 = transactionManager1.begin();
+
+  // Join the transaction begun by transactionManager1 with the transaction ID
+  transaction2 = transactionManager2.join(transaction1.getId());
+
+  // Execute CRUD operations in the transaction
+  Optional<Result> result = transaction1.get(...);
+  List<Result> results = transaction2.scan(...);
+  transaction1.put(...);
+  transaction2.delete(...);
+
+  // Prepare the transaction
+  transaction1.prepare();
+  transaction2.prepare();
+
+  // Validate the transaction
+  transaction1.validate();
+  transaction2.validate();
+
+  // Commit the transaction. If any of the transactions succeeds to commit, you can regard the
+  // transaction as committed
+  AtomicReference<Exception> exception = new AtomicReference<>();
+  boolean anyMatch =
+      Stream.of(transaction1, transaction2)
```

Comment:
With this error handling, I think the commits can be executed in parallel as follows

```java
  Stream.of(transaction1, transaction2)
    .parallel()
    .anyMatch(
        ...
```

Reply:
Ah, sorry. I found the parallel version code in the next section below.

This code is the sequential version, right? If so, please ignore my comment. But it might be good to mention sequential execution or something here?

Reply:
I think that has already been mentioned here:
https://github.com/scalar-labs/scalardb/pull/934/files#diff-2677f880f1a9e512f31cce8e12950863ddffd4be49408d310d72d4d73e6122b9R306

What do you think?

Reply:
Looks good!